python/kserve/test/test_infer_type.py

```diff
@@ -0,0 +1,344 @@
+                                              }]
+                                          )
+            res = infer_res.to_grpc()
+            assert res == expected
+
+        def test_from_grpc(self):
+            infer_res = ModelInferResponse(model_name="TestModel", id="123",
+                                           parameters={
+                                               "test-str": InferParameter(string_param="dummy"),
+                                               "test-bool": InferParameter(bool_param=True),
+                                               "test-int": InferParameter(int64_param=100)
+                                           },
+                                           outputs=[
+                                               {
+                                                   "name": "output-0",
+                                                   "shape": [1, 2],
+                                                   "datatype": "INT32",
+                                                   "contents": {
+                                                       "int_contents": [1, 2]
+                                                   },
+                                                   "parameters": {
+                                                       "test-str": InferParameter(string_param="dummy"),
+                                                       "test-bool": InferParameter(bool_param=True),
+                                                       "test-int": InferParameter(int64_param=100)
+                                                   },
+                                               }]
+                                           )
+            expected = InferResponse(model_name="TestModel", response_id="123",
+                                     parameters={
+                                         "test-str": InferParameter(string_param="dummy"),
+                                         "test-bool": InferParameter(bool_param=True),
+                                         "test-int": InferParameter(int64_param=100)
+                                     },
+                                     infer_outputs=[
+                                         InferOutput(name="output-0", datatype="INT32", shape=[1, 2], data=[1, 2],
+                                                     parameters={
+                                                         "test-str": InferParameter(string_param="dummy"),
+                                                         "test-bool": InferParameter(bool_param=True),
+                                                         "test-int": InferParameter(int64_param=100)
+                                                     })],
+                                     from_grpc=True
+                                     )
+            res = InferResponse.from_grpc(infer_res)
+            assert InferResponseMatcher(expected).__eq__(res) is True
```

Comment:
You should be able to use operator == after __eq__ is defined

Reply:
Expected == res

Reply:
done